test/csuite/time_shift_test.sh

```diff
@@ -0,0 +1,96 @@
+#! /bin/sh
+
+set -e
+
+# the purpose of this test is to ensure we use monotonic clock instead of
+# realtime clock in our code. we had the instances where WT is hanging when
+# system clock shifts (for eg: due to NTP servers). this test calculates
+# the execution time of a test(test_rwlock), shifts the clock -vely by that
+# time period and reexecutes the test. if the difference in the two execution
+# times is less than 20% test is considered passed. 20% is selected, based on
+# assumption that other factors of the environment will influence the execution
+# time by less than 20%.
+
+
+# need to enable long tests to run test_rwlock 
+export TESTUTIL_ENABLE_LONG_TESTS=1
+
+# We will run only when long tests are enabled.
+test "$TESTUTIL_ENABLE_LONG_TESTS" = "1" || exit 0
+
+EXIT_SUCCESS=0
+EXIT_FAILURE=1
+
+export DONT_FAKE_MONOTONIC=1
+RUN_OS=$(uname -s)
+
+# linux we run with cpu affinity, to control the execution time
+# if we don't control the execution time this test is not effective
+echo "test read write lock for time shifting using libfaketime"
+
+# check for the existence of dependent library
+if [ ! -r $1/libfaketimeMT.so.1 ]
+then
+   echo "fail : $1/libfaketimeMT.so.1 is not readable"
```

Comment:
Will this always fail with OS X now because of the different library name? I'd either change the input parameter to be the full library name (my preference) or remove OS X support below.